api/elastic.py

```diff
@@ -268,29 +275,47 @@ def identifierQuery(self, authorityList, authIdentText):
 
             if authority in authorityList:
 
-                return Q('bool', should=[
+                workAuthQuery = Q('bool', must=[
                     Q(
-                        'query_string',
+                        'term',
                         query=authority,
                         fields=['identifiers.authority'],
                         default_operator='and'
-                    ),
+                    )
+                ])
+
+                editionAuthQuery = Q('bool', must=[
                     Q(
-                        'query_string',
+                        'term',
+                        query=authority,
+                        fields=['editions.identifiers.authority'],
+                        default_operator='and'
+                    )
+                ])
+
+                workIdentQuery = Q('bool', must=[
+                Q(
+                    'term',
+                    query=identifier,
+                    fields=['identifiers.identifier'],
+                    default_operator='and'
+                )
+                ])
+
+                editionIdentQuery = Q('bool', must=[
+                    Q(
+                        'term',
                         query=identifier,
-                        fields=['identifiers.identifier'],
+                        fields=['editions.identifiers.identifier'],
                         default_operator='and'
                     ),
-                    Q(
-                        'nested',
-                        path='editions',
-                        query=Q(
-                            'query_string',
-                            query=identifier,
-                            fields=['editions.identifiers.identifier'],
-                            default_operator='and'
-                        )
-                    )
+                ])
+
+                return Q('bool', should=[
+                    Q('nested', path='identifiers', query=workAuthQuery),
+                    Q('nested', path='editions.identifiers', query=editionAuthQuery),
```

Comment:
These should be clustered in a nested `bool` query. That will give us only documents that match both authority and identifier. Right now this will give us _either_ authority or identifier matches.